Ana/add treasury proposals thresholds

Bitcoinera opened the pull request:
Closes #4988

**Description:**

<!-- Briefly describe what you're adding or fixing with this PR -->

All good

![image](https://user-images.githubusercontent.com/40721795/94148146-86d62a80-fe76-11ea-9b1c-4952d667d066.png)


I can make the threshold span more prominent though

![image](https://user-images.githubusercontent.com/40721795/94148194-99506400-fe76-11ea-9787-88c6803197bc.png)


EDIT: Threshold is not displayed anymore

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

faboweb commented:
<!-- NETLIFY DEPLOY COMMENT GENERATED BY ACTIONS_NETLIFY -->
🚀 Deployed on https://5f6c98f23f1f771a0115ba3c--lunieio.netlify.app

Bitcoinera commented:
WIP since there is a decision to be made on the display for Nays

faboweb commented on app/src/components/governance/ProposalStatusBar.vue:
```diff
@@ -26,13 +26,23 @@
-        :val="votePercentage * 100"
+        :val="
+          proposal.type === 'TREASURY'
+            ? proposal.detailedVotes.votingThresholdYes
```
this shows the threshold not the current status am I wrong? Shouldn't it be: yesVotes / votingThresholdYes?

Bitcoinera replied:
We could actually display both. But the latter is already visually displayed on the progress bar. But we could this display it as a percentage too :shrug: Not a bad idea

faboweb replied:
let's have a call on this

Bitcoinera replied:
Tuesday?

Bitcoinera replied:
oops, we forgot about this, didn't we?

jbibla commented:
this should be implemented as a bar for all proposals. but also, this is not a priority right now.

Bitcoinera commented:
> this should be implemented as a bar for all proposals. but also, this is not a priority right now.

The others don't have voting threshold, or you mean Yes/No bars (two bars instead of one)?

It might not be a priority right now but it would nice to wrap governance up, with the cherry on the top, don't you think?

jbibla commented:
everything has a threshold (some measure of when the state will change). for treasury proposals it's a majority of council members for example.

Bitcoinera commented:
> everything has a threshold (some measure of when the state will change). for treasury proposals it's a majority of council members for example.

In Polkadot referendums and treasuries, yes. But in the case of Cosmos I thought there is merely a deadline, and when this is reached it is decided upon the votes percentage at that point

jbibla commented:
> in the case of Cosmos I thought there is merely a deadline, and when this is reached it is decided upon the votes percentage at that point

yes, true. but at the deadline the state change is determined by the voting outcomes which we can display as a threshold.

Bitcoinera commented:
> yes, true. but at the deadline the state change is determined by the voting outcomes which we can display as a threshold.

The threshold is always simple majority in Cosmos, right?

jbibla commented:
> The threshold is always simple majority in Cosmos, right?

i don't know - please confirm with the cosmos docs!

Bitcoinera commented:
'kay, will have a look

Bitcoinera commented:
> please confirm with the cosmos docs!

It seems it is this way: https://docs.cosmos.network/v0.40/modules/gov/01_concepts.html#threshold

The only two things to take into account is that the proposal won't be valid if more than 1/3rd votes NoWithVeto and also a minimum Quorum must be reached. But we don't need to display that for now. Threshold is this what stands on the docs

Bitcoinera commented:
Seems there are still some issues with Polkadot proposals times :eyes: 

![image](https://user-images.githubusercontent.com/40721795/95578387-1a415b00-0a34-11eb-92f5-db950005e616.png)

Bitcoinera commented:
:thinking: 

![image](https://user-images.githubusercontent.com/40721795/95581193-def55b00-0a38-11eb-9cb8-2761f6b6d237.png)

 Guess we are missing something here

jbibla commented on app/src/components/governance/ProposalStatusBar.vue:
```diff
@@ -31,18 +31,14 @@
           <span v-if="voteCount">({{ voteCount }} Votes)</span>
         </div>
-        <div v-if="true" class="vote-data">
+        <div v-if="proposal.type === governanceStatusEnum.TREASURY" class="vote-data">
```
i don't understand why thresholds are only for treasury proposals...? every proposal type has a threshold.

Bitcoinera commented:
Alright, so I removed the simple display span for thresholds I added and this PR what is currently doing it simply fixing voting percentages in detailedVotes and in PropsalStatusBar

Bitcoinera commented:
Rerunning E2E

jbibla commented:
thanks ana!
